renovate/src/Renovate/Config.hs

```diff
@@ -110,28 +161,6 @@ data RenovateConfig arch binFmt (b :: * -> *) = RenovateConfig
   -- PowerPC safely.
   }
 
--- | The type of 'rcAnalysis'.
---
--- The motivation for analysis being in IO is to allow SFE to call out
--- to external tools, such as ILP solvers.
-type Analyze arch binFmt b = AnalyzeEnv arch -> MBL.LoadedBinary arch binFmt -> IO (b arch)
--- | The type of 'rcRewriter'.
-type Rewrite arch binFmt b = b arch
-                          -> MBL.LoadedBinary arch binFmt
-                          -> B.SymbolicBlock arch
-                          -> RW.RewriteM arch (Maybe [B.TaggedInstruction arch (B.InstructionAnnotation arch)])
```

Comment:
I think it's better to keep synonyms like this around, vs inlining their expansions. Here are some reasons:

- then synonym makes the code more readable. The `Rewrite arch binFmt b` is more meaningful/recognizable than `b arch -> MBL.LoadedBinary arch binFmt -> B.SymbolicBlock arch -> RW.RewriteM arch (Maybe [B.TaggedInstruction arch (B.InstructionAnnotation arch)])`, and this is true whether you've seen it before or not.
- using the synonym localizes type errors during refactoring. For example, if you change the definition of the synonym and the type of the identity transform is `identity :: Rewrite arch binFmt b`, then you get a type error at the definition site of `identity`. If you don't use the synonym, then you get a type error somewhere else, perhaps at the use site of `identity`, or even farther away if `identity` gets passed around (e.g. in a record) before use.

What's the motivation for inlining them?

Reply:
The following is subjective, of course.

My intent was to optimize for reading the code, rather than writing it.  My experience periodically coming back to this code has been frustration with the type synonyms - every time I saw one, I had to go look up what it was hiding.  This was especially acute for type synonyms over functions, where I had no idea what the types of arguments were.

There are a lot of different points in the design space, of course.  If the synonym were used many times in many places in such a way that was completely idiomatic that hid the fact that it was a type synonym (e.g., if we constructed all rewriters from a set of combinators where none of the arguments were ever visible), I think the synonym would be important to keep and an aid to readability.  In this case, most clients of this code have one or two instances, so the savings seem less compelling.